File: backend/app/routers/sessions.py
# backend/app/routers/sessions.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from typing import List
from app.db import get_db
from .. import models, schemas

router = APIRouter(prefix="/sessions", tags=["sessions"])

@router.post("", response_model=schemas.SessionOut)
def create_session(payload: schemas.SessionCreate, db: Session = Depends(get_db)):
    s = models.Session(**payload.model_dump())
    db.add(s); db.commit(); db.refresh(s)
    return s

@router.get("", response_model=List[schemas.SessionOut])
def list_sessions(db: Session = Depends(get_db)):
    return db.query(models.Session).order_by(models.Session.date.desc()).all()

File: frontend/src/api/sessions.ts
import { api } from "../api/client";

export type Session = {
  id: number; date: string; activity: string;
  location?: string; start_time?: string; end_time?: string;
};

export async function listSessions(): Promise<Session[]> {
  const { data } = await api.get("/sessions");
  return data;
}

export async function createSession(payload: {
  date: string; activity: string; location?: string;
  start_time?: string; end_time?: string;
}) {
  const { data } = await api.post("/sessions", payload);
  return data as Session;
}
